lib/kubernetes-deploy/kubernetes_resource/deployment.rb

```diff
@@ -54,13 +57,33 @@ def timeout_message
     end
 
     def deploy_timed_out?
-      super || @latest_rs && @latest_rs.deploy_timed_out?
+      if @progress
+        @progress["status"] == 'False'
+      else
+        super || @latest_rs && @latest_rs.deploy_timed_out?
```

Comment:
Sorry, I probably should have commented on `def timeout` instead. I was trying to say we should remove that override so that it always uses `TIMEOUT`. Here's an example of the scenario I'm worried about:
- spec.progressDeadlineSeconds is set to a low value, e.g. 30s
- the deployment has a ton of pods and a conservative rollout strategy, so it takes, say, 5min to finish
- partway through, for whatever reason, we don't observe the status condition (so `@progress` becomes nil)
- `deploy_timed_out?` sees `@progress == nil` and uses `super`, which thinks the timeout is 30s
- deploy times out even though it was still progressing, which is what we're trying to avoid with this feature

Of course landing in the `super` when progressDeadlineSeconds was set would be unexpected. But what I'm trying to say is if that happens for some reason, I think the sane behaviour would be to use the global timeout.